Fix window not resized when screen changes, #3715

low-batt opened the pull request:
The commit updates MainWindowController to resize the window
appropriately when the screen the window is on changes if in legacy
full screen mode.

The commit in the pull request will:
- Add a method setFrame to Extensions.NSScreen to set a window's
  frame and content view frame to use the full screen
- Add call to setFrame in observer for
  didChangeScreenParametersNotification

- [ ] This change has been discussed with the author.
- [x] It implements / fixes issue #3715.

---

**Description:**

lhc70000 commented on iina/Extensions.swift:
```diff
@@ -630,6 +630,22 @@ extension NSScreen {
+    // frame to avoid having part of the window obscured by the camera housing.
+    let view = window.contentView!
+    view.setFrameSize(NSMakeSize(view.frame.width, frame.height - unusable))
+  }
```
Here are my thoughts:

- It's redundant to have both `screen` and `window`. The screen should always be `window.screen ?? NSScree.main`; it makes little sense to call this method on a screen different from where the window is.
- We don't see much potential to use this method somewhere else. This shouldn't be a general method; Apart from setting the window frame, it also changes the content view, which is not expected for a method named `setFrame`.
- So, since we only use it in this specific case (legacy screen), it's better to rename the method to reflect the actual use and remove the argument. (Do we need to override `display` or `animate` in the future? If not, we can remove them.)
- If we expect the target window to be the `MainWindow`, it's better to move it to `MainWindowController`.

Therefore, I suggest putting it in `MainWindowController` with name
```swift
setWindowFrameForLegacyFullScreen()
```

Alternatively, if write as an extension, it should be an `NSWindow` extension, and its name should reflect the purpose:
```swift
window.setFrameAndContentViewToFullUseableSize()
```

low-batt commented:
I agree with all of that and I'm rather embarrassed as to how bad the proposed commit was. See if I properly addressed the comments in the new commit.

In testing this I sometimes noticed a lot of `didChangeScreenParametersNotification` notifications. That appears to be due to that notification being posted when macOS swaps the GPU handling the screen. I did not find a way to tell macOS to not notify us of a GPU change.
